### DISCUSSION TITLE
Compile from source with local mods

### DISCUSSION THREAD

**Discussion by nyholku** [Homebrew/discussions#1056] [Tap maintenance and Homebrew development]
Hi,

I have a program that can be installed succesfully from source with:

brew install --build-from-source gpsim

Now, I would like to make modifications to the source code and recompile, but don't know how to do that.

The problem I'm having is that if I do manually from Terminal what brew seems to be doing ie.

```
 ./configure --disable-gui --disable-shared --prefix=/usr/local/Cellar/gpsim/0.31.0
make all
```

Then the compilation fails.

Is the source code that brew uses somehow patched compared to the original source code?

If so where can I find the patch or patched source code?

**Comment by SMillerDev** [comment]
You can run `brew edit gpsim` to read what brew does to the code before installing.

**Comment by nyholku** [comment]
> On 16 Mar 2021, at 10.41, Sean Molenaar ***@***.***> wrote:
> 
> 
> You can run brew edit gpsim to read what brew does to the code before installing.
> 
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub, or unsubscribe.
> 

Thanks, good tip. Interesting that it appears that only a single trivial edit
that seems unrelated to my compilation failure is done.

I will keep investigating.

wbr Kusti

**Reply by carlocab** [reply]
`brew` also sets up environment variables to help with the build, and uses a compiler shim to intercept flags passed to the compiler and replace or supplement them with appropriate ones. The lack of those might also be your problem.

**Comment by nyholku** [comment]
> On 16 Mar 2021, at 11.08, Carlo Cabrera ***@***.***> wrote:
> 
> 
> brew also sets up environment variables to help with the build, and uses a compiler shim to intercept flags passed to the compiler and replace or supplement them with appropriate ones. The lack of those might also be your problem.
> 
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub, or unsubscribe.
> 

Ok. Tnx. Anyway I can see what is actually happening?

wbr Kusti

**Reply by carlocab** [reply]
You can try checking the logs. They're usually in `~/Library/Logs/Homebrew`.

**Comment by nyholku** [comment]
> On 16 Mar 2021, at 11.12, Carlo Cabrera ***@***.***> wrote:
> 
> 
> You can try checking the logs. They're usually in ~/Library/Logs/Homebrew.
> 
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub, or unsubscribe.
> 

Thanks, found them, and I see number of parameters added and some removed.

Not sure how I could apply that to my own source code.

How does formula specify what flags to use? 

Anyway, can I create a local formula that would compile from my local copy of the sources?

wbr Kusti

**Comment by ipatch** [comment]
you'd probably want to setup a tap for your formula because when you eventually update homebrew if changes to the formula your editing come down you'll end up with conflicts and could lose your changes.

i'd create a tap for your formula, make your changes and give your formula a different name than that of the standard formula ie. the name of the formula if it's in homebrew-core ie. `gdb` give it the name `my-gdb` as there are name conflicts if your tap/formula has that of the same name of a formula already in another tap.
